### PR TITLE
Add s390 testscripts by IBM

### PR DESCRIPTION
**Pull request by elifaln**
please add s390x_tests to the master branch

### CODE REVIEW

**Comment by okurz** [comment]
@Soulofdestiny

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    # Export the existing status of running tasks and system load for future reference (fail would export it again)
+    script_run "ps axf > /tmp/psaxf.log";
+    script_run "cat /proc/loadavg > /tmp/loadavg_consoletest_setup.txt";
+
+    # Just after the setup: let's see the network configuration
+    save_screenshot;
+
+    $self->clear_and_verify_console;
+
+
```
Comment (Soulofdestiny) [inline tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: Line 22 to Line 77 is not necessary, because we need to schedule `consoletest_setup.pm` before anyway, which does all these steps

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+# start of modification for s390 test TOOL_s390_vmcp    
+    my $TC_PATH = get_var('TC_PATH') . "/TOOL_s390_vmcp/";
```
Comment (Soulofdestiny) [inline tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: to be honest, I don't quite get what to set this variable to.
Was it some webserver you provided internally where you fetched the scripts from?
it might be worth to take a look on https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1779

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    assert_script_run "wget -r -np -R 'index.html*'  $TC_PATH";
+
+    script_run "cd $TC_PATH";
+    my $COMMONSH_PATH = get_var('TC_PATH') . "/lib/common.sh";
+    assert_script_run "wget $COMMONSH_PATH";
+    script_run "chmod +x ./*.sh";
+    assert_script_run "./vmcp_main.sh";
```
Comment (foursixnine) [inline tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: I wonder if it makes sense to actually save the output from assert_script_run "./vmcp_main.sh"; to a file, and then upload. @Soulofdestiny  wdyt?

Comment (Soulofdestiny) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: I didn't proceed that far yet, but might make sense yes =)

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 1 above)
Comment (elifaln) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: ok, we will remove this lines from consoletest_setup.pm.

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 2 above)
Comment (elifaln) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: We don’t know other means of transferring the test scripts to the target z/VM guest. Could you please suggest us the correct way of putting them? Is there preinstalled web server on openQA host which we can use? Or can we use “git fetch"?

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 3 above)
Comment (elifaln) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: Could you please help us to figure out how to properly upload gathered logs in openQA ?

File: tests/s390x_tests/TOOL_s390_vmcp/vmcp_main.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+section_start "Verify basic vmcp commands";
+verifyBasicVMCPCommands;
+section_end;
+
+section_start "Verify vmcp error conditions";
+verifyErrorConditions;
+section_end;
+
+show_test_results;
+section_end;
```
Comment (Soulofdestiny) [inline tests/s390x_tests/TOOL_s390_vmcp/vmcp_main.sh]: This function always return 0 - doesn't this overwrite the return value of "show_test_results" and leads to the whole testsuite always returning 0?

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 3 above)
Comment (foursixnine) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: @elifaln You can use the [upload_logs](http://open.qa/api/testapi/#_upload_logs ) method from the testapi.

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 2 above)
Comment (foursixnine) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: @elifaln if you put the data in the data directory of the tests, you can download them later via data_url($filename) call, like they're doing it in the [console/openvswitch_ssl](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/openvswitch_ssl.pm#L41) test

File: tests/s390x_tests/TOOL_s390_vmcp/vmcp_main.sh (hunk 4 above)
Comment (elifaln) [reply tests/s390x_tests/TOOL_s390_vmcp/vmcp_main.sh]: you are correct we will remove this function from test script.

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 3 above)
Comment (timbgr) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: `upload_logs`: good idea, we will use it in all our scripts to gather test output.
Thanks.

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 2 above)
Comment (timbgr) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: About `data_url`. We've tried the function and it provides some sophisticated (seems as one-time) URL, but we cannot figure out where actually to place our scripts into so they will be available via the provided URL? 

Could you please help us to figure out, which folder on an openQA server we must put our scripts into?

Comment (Soulofdestiny) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: Hi @timbgr 

so data_url() basically opens a connection on the worker from where the SUT (System under Test) can receive files from. the url contains a unique string bound to the job afaik.
The connection always points to the `data/` subdirectory of `os-autoinst-distri-opensuse`, so for example:

data_url(''): http://$worker:20993/C5mJaCtD3xFwm6YU/data/ points to https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/data
data_url('autoyast'): http://$worker:20993/C5mJaCtD3xFwm6YU/data/autoyast points to https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/data/autoyast


from here, the system under test could receive files e.g. via `curl` or `wget`.

Hope that answers your questions

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 1 above)
Comment (timbgr) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: We've tried to remove the lines from our modified `consoletest_vmcp.pm` but the test failed when it's being executed right after `consoletest_setup.pm`. 

It seems that in `consoletest_setup.pm` there are some lines which presumably stop some services and clean something up. In other scripts there is no such code. That is why when we delete the lines from `consoletest_vmcp.pm` according to the suggestion from the comment above, it stops working when being executed right after `consoletest_setup.pm` but it still works when being executed after other scripts like `consoletest_hyptop.pm`

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 2 above)
Comment (timbgr) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: I modified consoletest_vmcp.pm so now it's using data_url() function to fetch the shell scripts to be run.
`   script_run "mkdir TOOL_s390_vmcp";`
`    script_run "cd TOOL_s390_vmcp";`
`    assert_script_run "wget " . data_url('s390-tests/vmcp_main.sh');`
`    assert_script_run "wget " . data_url('s390-tests/common.sh');`
It's working fine :+1:

File: tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm (hunk 3 above)
Comment (timbgr) [reply tests/s390x_tests/TOOL_s390_vmcp/consoletest_vmcp.pm]: Modified `consoletest_vmcp.pm` to use upload_logs() function. 

`    assert_script_run "./vmcp_main.sh > vmcp_main.log";`
`    upload_logs('vmcp_main.log');`

Working fine :+1:

File: tests/s390x_tests/TOOL_s390_vmcp/vmcp_main.sh (hunk 4 above)
Comment (timbgr) [reply tests/s390x_tests/TOOL_s390_vmcp/vmcp_main.sh]: Removed the unnecessary `section_end` function.

Comment (foursixnine) [reply tests/s390x_tests/TOOL_s390_vmcp/vmcp_main.sh]: @timbgr I think you still need to push the changes :)

**Comment by timbgr** [comment]
We've updated the shell scripts and .pm script according to the comments above.

**Comment by Soulofdestiny** [comment]
I will work on the integration of this beginning next monday

File: README.md [hunk 5]
```diff
@@ -1,48 +0,0 @@
-os-autoinst/openQA tests for openSUSE and SUSE Linux Enterprise [![Build Status](https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse.svg?branch=master)](https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse?branch=master)
```
Comment (asmorodskyi) [inline README.md]: what the reason to delete readme ?

Comment (elifaln) [reply README.md]: Hi, something unintentional happen. I’m sorry for that, it was not intentional to delete this README file.

File: tests/s390x_tests/lib/common.sh [hunk 6]
```diff
@@ -1,999 +0,0 @@
-# Copyright (C) 2018 IBM Corp.
```
Comment (Soulofdestiny) [inline tests/s390x_tests/lib/common.sh]: quick question - why were those files removed?

Comment (elifaln) [reply tests/s390x_tests/lib/common.sh]: Hi Matthias, we splitted this script to small scripts (https://github.com/elifaln/os-autoinst-distri-opensuse/tree/s390x_tests/tests/s390x_tests/lib).

File: tests/s390x_tests/README.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# s390x tests for openQA 
+
+This repository contains s390x specific test cases which can be executed with openQA. 
+Each test case contains a README file, test scripts and a consoletest_xxx.pm.
+
+## Getting started
+Each consoltest will be included in main.pm. The consoltests are fetched and executed by the test scripts.
```
Comment (okurz) [inline tests/s390x_tests/README.md]: s/consoltest/consoletest/

File: tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh [hunk 8]
```diff
@@ -0,0 +1,199 @@
+# Copyright (C) 2018 IBM Corp.
+# 
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+
+#!/bin/bash
```
Comment (okurz) [inline tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh]: It is very uncommone to have a shebang not in the first line _ if this actually works ;)

I also recommend to call `#!/bin/sh -e` to not ignore errors. If you actually need bash and not just POSIX sh then of course `#!/bin/bash -e`

File: tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+# initially basic options of both commands are tested (help menu and
+# tool version). Then, using chchp, a channel path given as parameter
+# is disabled, enabled, deconfigured and configured. lschp is used to
+# check if the channel path current state is the expected one.
+# This script if the basic options of chchp are working properly and
+# disables, enables, deconfigures and configures a channel path using
+# chchp. Then verifies the result with lschp.
+
+source "lib/auxx.sh" || exit 1
+source "lib/env.sh" || exit 1
```
Comment (okurz) [inline tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh]: See above, with `-e` this should not be necessary.

File: tests/s390x_tests/TOOL_s390_chchp/consoletest_TOOL_s390_chchp.pm [hunk 10]
```diff
@@ -0,0 +1,38 @@
+# SUSE’s openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC                                                                                    
+# Copyright (C) 2018 IBM Corp.
```
Comment (okurz) [inline tests/s390x_tests/TOOL_s390_chchp/consoletest_TOOL_s390_chchp.pm]: It's ok to just use the IBM copyright from 2018. I do not see any SUSE content left ;)

File: tests/s390x_tests/lib/net_ifup.sh [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+    #ifconfig $IFNAME
+    ip addr show $IFNAME
+    echo
+}
+
+
+
+
+
+
```
Comment (okurz) [inline tests/s390x_tests/lib/net_ifup.sh]: Please remove blank lines at end-of-line

File: tests/s390x_tests/lib/s390base.pm [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+sub cleanup_testsuite {
+    return 1;
+    # FIXME assert_script_run 'cd / && rm -rf ./tmp';
+}
+
+
+1;
+# vim: set sw=4 et:
```
Comment (okurz) [inline tests/s390x_tests/lib/s390base.pm]: By now we removed this boiler-plate code in all template files. So this line can be safely removed as well.

File: tests/s390x_tests/lib/s390base.pm [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+sub execute_script {
+    my ($self, $script, $scriptargs, $timeout) = @_;
+    assert_script_run ("./$script $scriptargs  >> $script.log 2>&1",  timeout => $timeout);
+    save_screenshot;
+    upload_logs "$script.log";
+}
+
+sub cleanup_testsuite {
+    return 1;
+    # FIXME assert_script_run 'cd / && rm -rf ./tmp';
```
Comment (okurz) [inline tests/s390x_tests/lib/s390base.pm]: Please do not add disabled code as nobody but you will understand why this is not active. Better leave it out, potentially keep in your local git repository, either in `git stash` or a temporary "WIP"-commit

File: tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+	# Use --vary and -v
+	echo "Turn off CHPID $chpid_1 using Vary"
+	assert_exec 0 "chchp --vary 0 $chpid_1"
+	if [[ "$(chp::getVary "${chpid_1}")" -eq 0 ]]; then
+		echo "CHPID $chpid_1 vary OFF was successful"
+	else
+		echo "CHPID vary OFF failed"
+	fi
+
+	sleep 5
```
Comment (asmorodskyi) [inline tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh]: "magic number" - better to extract to a variable

File: tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh (hunk 8 above)
Comment (timbgr) [reply tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh]: Ok, we move shebang to the top :) it works anyway probably because the script is not executed from a console but is being called by other script.

As for "-e" key, I've tried it and it has broken all our script because scripts are written to catch and handle errors when necessary, counting successes and failures. If we add "-e" key, then the script just exits at the first error, even if the error is expected by our its logic (e.g. checks that expects errors to happen, negative checks and so on). That is why better leave it without "-e" key.

Comment (okurz) [reply tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh]: ok, I see. Thank you for looking into this.

**Comment by okurz** [comment]
I am sorry if I am bringing this up late but I see that the commit subject lines do not really stick to what we have as a common practice in this repo. Please start the git commit message subject lines with a capital letter, and phrase them in imperative mood, e.g. "Add new test case MEMORY_memtester" instead of "added new Test case …". Also see http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages. Of course, an easy alternative is to [squash commits](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history) and just create one with the right name :)

File: tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh (hunk 14 above)
Comment (timbgr) [reply tests/s390x_tests/TOOL_s390_chchp/chchpmain.sh]: Ok, changed that to a variable

File: tests/s390x_tests/lib/main_common.pm [hunk 15]
```diff
@@ -0,0 +1,1909 @@
...
+    if (get_var("ADDONS", "") =~ /rt/) {
+        loadtest "rt/kmp_modules";
+    }
+
+    loadtest "console/consoletest_setup";
+
+    if (get_var('IBM_TESTS')) {
+        # prepare tarballs for the testcase
+        # the path below should be reworked to be universal for any distribution, now it's for openQA deployed on opensuse 
+        my $tcs_path = "/var/lib/openqa/share/tests/sle/data/s390x/";
```
Comment (Soulofdestiny) [inline tests/s390x_tests/lib/main_common.pm]: I think those pathes are not correct anymore, no?

File: tests/s390x_tests/lib/main_common.pm [hunk 16]
```diff
@@ -0,0 +1,1909 @@
...
+    loadtest "console/ssh_pubkey";
+    loadtest "console/ssh_cleanup";
+    loadtest "console/openvswitch_ssl";
+    loadtest "console/consoletest_finish";
+}
+
+sub load_security_tests_web {
+    loadtest "console/curl_https";
+    loadtest "console/wget_https";
+    loadte
```
Comment (Soulofdestiny) [inline tests/s390x_tests/lib/main_common.pm]: This main_common.pm should be in `tests/sle/lib/` and not in s390x subfolder

File: tests/s390x_tests/lib/main_common.pm (hunk 15 above)
Comment (timbgr) [reply tests/s390x_tests/lib/main_common.pm]: The path `my $tcs_path = "/var/lib/openqa/share/tests/sle/data/s390x/";`
is exactly what we have in our openQA server which is based on openSuSE
so in our environment it's correct path
```
osus2:/var/lib/openqa/share/tests/sle/data/s390x # ls
TOOL_s390_CHCHP  TOOL_s390_ZIOMON  TOOL_s390_fcp  TOOL_s390_hyptop  TOOL_s390_qethqoat  TOOL_s390_vmcp  lib

osus2:/var/lib/openqa/share/tests/sle/data/s390x # cat /etc/os-release
NAME="openSUSE Leap"
VERSION="42.3"
...

osus2:/var/lib/openqa/share/tests/sle/data/s390x # ls -la /var/lib/openqa/share/tests
total 12
drwxr-xr-x  3 geekotest root 4096 Oct  2 03:10 .
drwxr-xr-x  4 root      root 4096 Aug  6 08:07 ..
drwxr-xr-x 10 geekotest www  4096 Oct  2 03:10 opensuse
lrwxrwxrwx  1 root      root    9 Jul 17 07:15 sle -> opensuse/

```

File: tests/s390x_tests/lib/main_common.pm (hunk 16 above)
Comment (timbgr) [reply tests/s390x_tests/lib/main_common.pm]: Yep, correct. We've just put main_common.pm to this folder temporarily as we change only one function in `main_common.pm`. We suppose that you can extract only this function 
`sub load_consoletests` and put it into the master branch main_common.pm

But we can put our versions of both `main_common.pm` and `s390base.pm` into their **real** folders so you can try merging our changes into master automatically? is it ok?

Comment (Soulofdestiny) [reply tests/s390x_tests/lib/main_common.pm]: > But we can put our versions of both `main_common.pm` and `s390base.pm` into their **real** folders so you can try merging our changes into master automatically? is it ok?

That's the most stable solution I guess

Comment (timbgr) [reply tests/s390x_tests/lib/main_common.pm]: Ok, we've moved all the test shell scripts and consoletest_ perl modules to the appopriate locations. Now they are exactly the same as on our test servers where testcases are currently executed.

Here are the shellscripts and lib files and config files: https://github.com/elifaln/os-autoinst-distri-opensuse/tree/s390x_tests/data/s390x
Here are the consoletest perl modules: https://github.com/elifaln/os-autoinst-distri-opensuse/tree/s390x_tests/tests/s390x_tests

**Comment by timbgr** [comment]
common.sh is actually deleted and should not be referenced in any script

it;s been replaced by set of libraries like `auxx.sh`, `env.sh` and so on

**Comment by Soulofdestiny** [comment]
@elifaln @timbgr 
could you please try to rebase this PR on current master? with squashed commits (in logical groups)

**Comment by timbgr** [comment]
Ok. We've done the rebase and now I see there is no more conflicts in `main_common.pm`

**Comment by Soulofdestiny** [comment]
looks good so far.

one issue I found is, that we have issues with serial console on zVM, I have a workaround for that - it makes sense to put this into this PR as well
it's in `console/consoletest_setup.pm`, we need to exclude the stop/disable of getty for s390x backend
so please add this snippet to your code:

```
 30     if (!check_var('BACKEND', 's390x')) {
 31         systemctl "stop serial-getty\@$testapi::serialdev", ignore_failure => 1;
 32         systemctl "disable serial-getty\@$testapi::serialdev";
 33         # Mask if is qemu backend as use serial in remote installations e.g. during reboot
 34         systemctl "mask serial-getty\@$testapi::serialdev" if check_var('BACKEND', 'qemu');
 35     }
```

L30 and L35 were added - that's all

File: tests/s390x_tests/consoletest_MEMORY_chmem.pm [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+sub run {
+    my $self = shift;
+    $self->copy_testsuite('MEMORY_chmem');
+    $self->execute_script('chmem.test.sh','1', 1200 );
+    $self->execute_script('checksum.chmemtest.sh','1', 300 );
+}
+
+sub post_fail_hook {
+    my $self = shift;
+#    $self->export_logs();
```
Comment (Soulofdestiny) [inline tests/s390x_tests/consoletest_MEMORY_chmem.pm]: This test fails at the moment - but I cannot see why because this line is commented

File: tests/s390x_tests/consoletest_KERNEL_btrfs.pm [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+
+sub run {
+    my $self = shift;
+    $self->copy_testsuite('KERNEL_btrfs');
+    $self->execute_script('btrfs.eckd.sh', '"xxxx xxxx xxxx xxxx xxxx xxxx"', 1200);
+}
+
+sub post_fail_hook {
+    my $self = shift;
+   $self->export_logs();
```
Comment (Soulofdestiny) [inline tests/s390x_tests/consoletest_KERNEL_btrfs.pm]: somehow logs for this test are not uploaded... not sure why exactly - other logs are uploaded correctly

File: tests/s390x_tests/consoletest_KERNEL_LSCPU_CHCPU.pm [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+    my $self = shift;
+    $self->copy_testsuite('KERNEL_LSCPU_CHCPU');
+    $self->execute_script('test_lscpu.sh','', 600 );
+    $self->execute_script('test_chcpu.sh','', 600 );
+    $self->execute_script('test_lscpu_chcpu_invalid.sh','', 300);
+    $self->execute_script('lscpu_chcpu_kernel_parm.sh', 'start', 300);
+    $self->execute_script('lscpu_chcpu_kernel_parm.sh', 'test1', 300);
+    $self->execute_script('lscpu_chcpu_kernel_parm.sh', 'test2', 300);
+    
+    power_action('reboot', observe => 1, keepconsole => 1);
```
Comment (Soulofdestiny) [inline tests/s390x_tests/consoletest_KERNEL_LSCPU_CHCPU.pm]: This function needs to be imported first:
`# Test died: Undefined subroutine &consoletest_KERNEL_LSCPU_CHCPU::power_action called at /var/lib/openqa/share/tests/sle/tests/s390x_tests/consoletest_KERNEL_LSCPU_CHCPU.pm line 30.
`

**Comment by timbgr** [comment]
Ok we'll include the  two lines  into consoletest_setup.pm

File: tests/s390x_tests/consoletest_TOOL_s390_vmconvert.pm [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+use base "s390base";
+use testapi;
+use utils;
+use strict;
+
+sub run {
+    my $self = shift;
+    $self->copy_testsuite('TOOL_s390_vmconvert');
+    
+    my $REPO_SUSE = get_var('REPO_SUSE');
```
Comment (Soulofdestiny) [inline tests/s390x_tests/consoletest_TOOL_s390_vmconvert.pm]: using `get_required_var` is better here, since it's more obvious to the user why it's failing

**Comment by timbgr** [comment]
About ZIOMON testcase, the following parameters must be used, pointing to some (to be erased and formatted) disk:
PARM_ADAPTER= 
PARM_WWPN= 
PARM_LUN=

**Comment by elifaln** [comment]
Hi Matthias, consoletest_setup.pm is updated as discuss. We took the most recent version from os-autoinst:master  branch, added the lines you suggested and pushed the file.

File: tests/s390x_tests/consoletest_KERNEL_LSCPU_CHCPU.pm (hunk 19 above)
Comment (vkpro) [reply tests/s390x_tests/consoletest_KERNEL_LSCPU_CHCPU.pm]: Please inform how this function can be imported?
I use `use utils` and it works very well in my env.

File: tests/s390x_tests/consoletest_TOOL_s390_vmconvert.pm (hunk 20 above)
Comment (vkpro) [reply tests/s390x_tests/consoletest_TOOL_s390_vmconvert.pm]: Replaced with `get_required_var `

File: tests/s390x_tests/consoletest_KERNEL_btrfs.pm (hunk 18 above)
Comment (vkpro) [reply tests/s390x_tests/consoletest_KERNEL_btrfs.pm]: It works in my env. Please provide test case logs from the test system.

File: tests/s390x_tests/consoletest_MEMORY_chmem.pm (hunk 17 above)
Comment (vkpro) [reply tests/s390x_tests/consoletest_MEMORY_chmem.pm]: Fixed

**Comment by Soulofdestiny** [comment]
I guess the small thing which is adding the conflict, is the split of consoletest_setup.pm
so take a look into https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6032 which might help you here

**Comment by elifaln** [comment]
Hi Matthias, we just took the most recent versions of the two files from `master` branch, and added our changes there. But that not help to solve the conflict, have any idea how to solve?

**Comment by okurz** [comment]
Sorry but it seems your last changes caused other conflicts now. It seems the "changed files" show something that you did not really want to change

**Comment by elifaln** [comment]
The remaining conflicts in the two files seem to be exactly the differences which we need to add. In consoletest_setup.pm those are the lines suggested by Matthias. In main_common.pm are also the lines which are needed for s390x_tests to be scheduled.

**Comment by Soulofdestiny** [comment]
> The remaining conflicts in the two files seem to be exactly the differences which we need to add. In consoletest_setup.pm those are the lines suggested by Matthias. In main_common.pm are also the lines which are needed for s390x_tests to be scheduled.

hey @elifaln, happy new year first of all!
any news on this? are you able to resolve those conflicts?

**Comment by Soulofdestiny** [comment]
this should do the trick - let's wait what travis says and then one can squash and rewrite the commit message

**Comment by Soulofdestiny** [comment]
mhh I still see some issues, let me check

**Comment by Soulofdestiny** [comment]
@elifaln I'm running this PR right now against the latest SLE15-SP1 build. I'll let you know the results
but it would be nice to clean the commit message and state why we are doing this instead of listing all the "add test xy".

**Comment by okurz** [comment]
@Soulofdestiny further test failures: https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse/jobs/501526658#L728

I suggest to run `make test` locally to find all problems to have faster feedback.
